src/light/AmbientLight.ts

```diff
@@ -9,21 +9,28 @@ import { PRODUCT_LAYER_MASK } from '../constant/VisibilityLayerMask';
  * @module
  */
 
-export default class DIVEAmbientLight extends AmbientLight {
+export default class DIVEAmbientLight extends Object3D {
+    private _light: AmbientLight;
+
     constructor() {
-        super(0xffffff, 1);
-        this.layers.mask = PRODUCT_LAYER_MASK;
+        super();
+
+        this.name = 'DIVEAmbientLight';
+
+        this._light = new AmbientLight(0xffffff, 1);
+        this._light.layers.mask = PRODUCT_LAYER_MASK;
+        this.add(this._light);
     }
 
     public SetColor(color: Color): void {
-        this.color = color;
+        this._light.color = color;
     }
 
     public SetIntensity(intensity: number): void {
-        this.intensity = intensity;
+        this._light.intensity = intensity;
     }
 
     public SetEnabled(enabled: boolean): void {
-        this.visible = enabled;
+        this._light.visible = enabled;
```

Comment:
Enabled and Visible could mean different things, perhaps you could use a unique term?

Reply:
Since in the case of AmbientLight and SceneLight we don't really differentiate between 'visible' and 'enabled' when we look at the result.

The difference is more clear when you look at [PointLight:SetEnabled](https://github.com/shopware/dive/blob/6a1e8f63c914125a3b8c89e0722e3abe7e5e0420/src/light/PointLight.ts#L65) and [LightRoot:UpdateLight](https://github.com/shopware/dive/blob/6a1e8f63c914125a3b8c89e0722e3abe7e5e0420/src/scene/root/lightroot/LightRoot.ts#L64):

In SetEnabled we set the light itself to visible: true/false. This results in the PointLight's second child (the physical representation aka Mesh) still being visible.

In UpdateLight we set the wrapper classes visibility to true/false which includes the physical representation as well.

This results in a functional difference of switching a light on or off and showing or hiding a light completely within the scene.

Does this make sense to you?

Reply:
No, but trust you :) ok ok